### DISCUSSION TITLE
Phazerville Suite v1.6

### DISCUSSION THREAD

**Discussion by djphazer** [djphazer/O_C-Phazerville#11] [Announcements]
It's probably not perfect, but it's a level up!

### New Stuff
In Hemisphere, Long-press DOWN button for the new Config Menu:
* **4 preset banks** for hot-swapping applet configs.
* When powered on, _Preset A_ is loaded by default (if valid)
* (NOTE: Only the presets are stored in EEPROM now. You must store the state in a preset before doing the double-right-encoder-long-press thing.)
* There are also settings for **Edit Cursor Mode** and **Trigger Pulse Length**, stored per preset.

More improvements to the internal Clock:
* The clock state now cycles from **Stop->Pause->Start->Stop**
* **Sync-Start** mechanism - When Paused, the first pulse on TR1 will Start the Clock. Subsequent pulses on TR1 will be interpreted based on the PPQN setting for external tempo sync like usual.
  * It _does not_ automatically stop (like PNW Clk input) but this is a goal...
* There's a new **manual tap-tempo** feature hidden in Clock Setup - 4 taps on the Tempo parameter will actually set the BPM accordingly.

I also restored the **[Game Of Life](https://github.com/Chysn/O_C-HemisphereSuite/wiki/Conways's-Game-of-Life-(Retired))** applet because it's just so neat!

### Build Choices
All of the original O_C apps (sans _References_) are in here now, available in 2 alternative builds.
All 3 builds include **Hemisphere**, **Low-rents**, and **Pong**, plus:
* Main builds = **Calibr8or, Quadraturia, Piqued, Enigma, Captain MIDI, Neural Net, Darkest Timeline**
* The "+stock" builds both include: **CopierMaschine, Quantermain, Acid Curds, Automatonnetz**, and:
  - _+stock1_ = **Piqued, Sequins**
  - _+stock2_ = **Quadraturia, Meta-Q, Harrington 1200, Dialectic Ping Pong, Viznutcracker**

### Continuous Improvement
The preset switching functionality is likely to expose some quirks about how various applets save/load state. If you notice something that could be improved, I welcome bug reports and PRs! And again, if you want a specific combo/configuration that I haven't provided, let me know, or modify `platformio.ini` and build it yourself! ;)

After this, I expect to spend less time on development until after summer... There are some applet tweaks and other ideas I want to do for 1.6.x, but I also want to do a major overhaul to the internal class/data structures for v2.0, to pave the way for bigger ideas! Unified quantizer settings, more flexible MIDI handling, I/O routing between applets... stay tuned. ⚡

<hr /><em>This discussion was created from the release <a href='https://github.com/djphazer/O_C-BenisphereSuite/releases/tag/PSv1.6'>Phazerville Suite v1.6</a>.</em>

**Comment by toornsthulpft** [comment]
Thank you very much for your work on this! It's a fantastic happy medium between stock and hemisphere. Is there any way to create a hex file for a flipped VOR and/or "stock+" VOR variant? Thanks again!

**Reply by djphazer** [reply]
Ok, I can do that! I was wondering if any OCP users liked to run their module "upside down" 😉

So many permutations to maintain now, lol... maybe some of them could be reimplemented as runtime options 🤔

**Reply by djphazer** [reply]
All VOR/flipped variants are attached to the release now. All untested! *nervous chuckle* Good luck!

**Reply by toornsthulpft** [reply]
Thank you so much! It occurred to me afterward that the attenuverter behavior might end up wonky with a flipped orientation, but I'll give it a shot.

Again, really appreciate your work here. It's by far my favorite fw for the o_c.

**Comment by vestedaf** [comment]
New version! Woo! Loading tonight and I'll let you know if I'm still having issues saving in Acid Curds like I was in the previous version.

**Comment by bennyblancodisco** [comment]
Just happened to come across this new flavor of O&C firmware and OMG.... such a great merging of apps and so jammed back. I've flashed 4 O&C (2 plum, 2 talldog 8 hp). I had been waiting on Bennisphere to get to VOR but just hasn't happened.... and then this!
BOOM. Time to start patching!

Thanks a bunch for everyone's efforts that contributed to this branch.

**Comment by fragletrollet** [comment]
Just wanted to add my appreaciation of this fork. All my O&C's are running 1.6 with and without VOR and working great. So nice to have Sequins and Piqued available together with the Beni suite.

Keep up the great work! <3

**Comment by swordsreversed** [comment]
Hello is anyone have an issue in flipped where the knobs go in the opposite direction? Any ideas on how to fix this would be welcome...

**Reply by djphazer** [reply]
There is a calibration setting...

**Reply by swordsreversed** [reply]
Ok i'll try and find it...

**Reply by djphazer** [reply]
Sorry I didn't have time to explain earlier... I assume you have an 8HP uO_C? I tried to make this automatic, but I think some hardware has different encoder layout anyway, so there's no universal solution.

Go to the "Setup / About" app, push left encoder to enter [CALIBRATE] procedure. On the first page, you'll see "Encoders: R reversed" or similar. Push either UP or DOWN buttons to cycle through. Go forward to end of calibration to save.

**Comment by swordsreversed** [comment]
Thanks so much! I was struggling to find where the setup was. All good now.
